packages/jsii-pacmak/lib/targets/dotnet.ts

```diff
@@ -126,44 +127,845 @@ export default class Dotnet extends Target {
+                this.code.line('return InvokeInstanceMethod<' + returnType + '>(new object[]{});');
+            } else {
+                this.code.openBlock(`${access} ${staticKeyWord}${overrideKeyWord}${virtualKeyWord}${signature}`);
+                this.code.line('InvokeInstanceVoidMethod(new object[]{});');
+            }
+            this.code.closeBlock();
+        }
+    }
+
+    // Founds out if a member (property or method) is already defined in one of the base classes
+    // Used to figure out if the override or virtual keywords are necessary.
+    private isMemberDefinedOnAncestor(cls: spec.ClassType, member: spec.Property | spec.Method): boolean {
+        if (member as spec.Method) {
+            const objectMethods = [ 'ToString', 'GetHashCode', 'Equals'];
+            // Methods defined on the Object class should be overridden, return true;
+            if (objectMethods.includes(nameutils.convertMethodName(member.name))) {
+                return true;
+            }
+        }
+
+        const base = cls.base;
+        if (base) {
+            const baseType = this.findType(base) as spec.ClassType;
+
+            if (member as spec.Property) {
+                if (baseType.properties) {
+                    if (baseType.properties.filter(property => property.name === member.name).length > 0) {
+                        // property found in base parent
+                        return true;
+                    }
+                }
+                return this.isMemberDefinedOnAncestor(baseType, member);
+            } else if (member as spec.Method) {
+                if (baseType.methods) {
+                    if (baseType.methods.indexOf(member) > 0) {
+                        // Method found in base parent
+                        return true;
+                    }
+                }
+                return this.isMemberDefinedOnAncestor(baseType, member);
+            } else {
+                return false;
+            }
+        } else {
+            return false;
+        }
+    }
+
+    // Emits all documentation depending on what is available in the jsii model
+    // Used by all kind of members + classes, interfaces, enums
+    private emitDocs(obj: spec.Method | spec.InterfaceType | spec.ClassType | spec.Property | spec.EnumType): void {
+        if (obj.docs) {
+            if (obj.docs!.summary) {
+                this.code.line(`/// <summary>${obj.docs!.summary}</summary>`);
+            }
+            if (obj.docs!.remarks) {
+                this.code.line(`/// <remarks>`);
+                const remarkLines = obj.docs!.remarks.split('\n');
+                remarkLines.forEach( line => this.code.line(`/// ${line}`));
+                this.code.line(`/// </remarks>`);
+            }
+            if (obj.docs!.returns) {
+                this.code.line(`/// <returns>${obj.docs!.returns}</returns>`);
+            }
```

Comment:
There's definitely some behavior that has disppeared here.

* Examples
* Parameter docstrings
* Arbitrary annotations (links, deprecated, etc.)

Reply:
I have added the full documentation generation. Not all of it is required by the models currently generated but it should be there for the big *calc* model